commcare_connect/events/models.py

```diff
@@ -0,0 +1,143 @@
+from abc import ABCMeta, abstractproperty
+from dataclasses import dataclass
+from datetime import datetime
+
+from django.db import models
+from django.utils.translation import gettext as _
+
+from commcare_connect.cache import quickcache
+from commcare_connect.users.models import User
+
+from . import types
+
+
+class Event(models.Model):
+    from commcare_connect.opportunity.models import Opportunity
+
+    # this allows referring to event types in this style: Event.Type.INVITE_SENT
+    Type = types
+
+    date_created = models.DateTimeField(db_index=True)
+    event_type = models.CharField(max_length=40, db_index=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE, null=True)
+    opportunity = models.ForeignKey(Opportunity, on_delete=models.PROTECT, null=True)
```

Comment:
I will add again that I think a metadata field will be very beneficial. Even with the existing set of events we have, data like which record was approved, or how much the payment was could be very useful, and I know that mobile was also hoping to include additional metadata.